replica-common/src/main/java/com/graphhopper/reader/osm/CustomOsmReader.java

```diff
@@ -368,7 +371,13 @@ else if (!config.getElevationSmoothing().isEmpty())
         setArtificialWayTags(pointList, way, distance, nodeTags);
         IntsRef relationFlags = getRelFlagsMap(way.getId());
         EdgeIteratorState edge = baseGraph.edge(fromIndex, toIndex).setDistance(distance);
+
+        // Set SEID for edge in both directions before Way tags are handled, to ensure SEID-based custom speeds can be applied
+        // SEIDs require OSM IDs to be calculated, so we must parse those first
+        osmParsers.handleOsmIdWayTags(edge.getEdge(), edgeIntAccess, way, relationFlags);
+        stableEdgeIdManager.setStableEdgeIdsForEdge(edge, segmentIndex);
         osmParsers.handleWayTags(edge.getEdge(), edgeIntAccess, way, relationFlags);
```

Comment:
`handleWayTags` is where we'll apply user-provided SEID-based custom speeds, in a future PR